frontend/libs/studio-components/src/components/StudioNumberInput/utils.test.ts

```diff
@@ -0,0 +1,19 @@
+import { convertNumberToString, convertStringToNumber, isStringValidDecimalNumber } from './utils';
+
+describe('utils', () => {
+  it('isStringValidDecimalNumber should return true when input is integer', () => {
+    expect(isStringValidDecimalNumber('123')).toBe(true);
+  });
+
+  it('isStringValidDecimalNumber should return true when input is decimal', () => {
+    expect(isStringValidDecimalNumber('123.456')).toBe(true);
+  });
+
+  it('convertStringToNumber should return number', () => {
+    expect(convertStringToNumber('123.456')).toBe(123.456);
+  });
+
+  it('convertNumberToString should return string', () => {
+    expect(convertNumberToString(123.456)).toBe('123,456');
+  });
+});
```

Comment:
Fine tester! Vi bør imidlertid teste flere forskjellige situasjoner, og særlig spesielle situasjoner som når inndataen er `''`, `0`, `undefined` osv. Da trenger vi heller ikke ha like detaljerte tester på selve komponenten. Vi bør også ha en `describe`-blokk per funksjon for å lettere kunne se hva testene tester. Her er et forslag:
```suggestion
describe('utils', () => {
  describe('isStringValidDecimalNumber', () => {
    it('should return true when input is integer', () => {
      expect(isStringValidDecimalNumber('123')).toBe(true);
      expect(isStringValidDecimalNumber('0')).toBe(true);
    });

    it('should return true when input is decimal', () => {
      expect(isStringValidDecimalNumber('123.456')).toBe(true);
      expect(isStringValidDecimalNumber('123,456')).toBe(true);
    });

    it('should return false when input is not a number', () => {
      expect(isStringValidDecimalNumber('abc')).toBe(false);
      expect(isStringValidDecimalNumber('123abc')).toBe(false);
      expect(isStringValidDecimalNumber('')).toBe(false);
    });
  });

  describe('convertStringToNumber', () => {
    it('should return number', () => {
      expect(convertStringToNumber('123.456')).toBe(123.456);
      expect(convertStringToNumber('123,456')).toBe(123.456);
      expect(convertStringToNumber('123')).toBe(123);
      expect(convertStringToNumber('')).toBe(0);
    });
  });

  describe('convertNumberToString', () => {
    it('should return string', () => {
      expect(convertNumberToString(123.456)).toBe('123,456');
      expect(convertNumberToString(123)).toBe('123');
      expect(convertNumberToString(0)).toBe('0');
      expect(convertNumberToString(undefined)).toBe('');
      expect(convertNumberToString(null)).toBe('');
    });
  });
});
```
Når man skriver tester på denne måten, er det lett å oppdage feil og ting som kanskje burde være annerledes. Her ser vi f.eks. at `convertStringToNumber('')` blir `0`. Gir dette egentlig mening? Kanskje den heller burde returnere `undefined`? Da unngår vi feil som den Finnur påpekte i sted.